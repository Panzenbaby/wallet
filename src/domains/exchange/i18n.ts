export const translations: { [key: string]: any } = {
	EXCHANGE_NAME: "Exchange Name",

	MODAL_ADD_EXCHANGE: {
		TITLE: "Exchange",
	},

	PAGE_EXCHANGES: {
		ADD_EXCHANGE: "Add Exchange",
		EMPTY_MESSAGE: "There are currently no exchange providers available, please try again at a later time.",
		SUBTITLE: "Choose one of the exchange providers below to swap directly from within the wallet.",
		TITLE: "Exchange",
	},

	SELECT_EXCHANGE_MESSAGE: "Select one of the exchanges you have installed",

	YOUR_EXCHANGE_LIST: "Your Exchange List",
};
