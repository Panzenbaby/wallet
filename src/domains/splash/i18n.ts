export const translations: { [key: string]: any } = {
	BRAND: "Payvo Wallet",
	COPYRIGHT: "Â© ARK.io",
	LOADING: "Initializing...",
	PRODUCT: "An ARK.io Product",
	RIGHTS: "All Rights Reserved",
	VERSION: "Version",
};
