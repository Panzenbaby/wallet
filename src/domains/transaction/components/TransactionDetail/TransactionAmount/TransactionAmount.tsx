import { Amount, AmountCrypto } from "app/components/Amount";
import { Circle } from "app/components/Circle";
import { Icon } from "app/components/Icon";
import { Label } from "app/components/Label";
import { Tooltip } from "app/components/Tooltip";
import cn from "classnames";
import React from "react";
import { useTranslation } from "react-i18next";

import { TransactionDetail, TransactionDetailProperties } from "../TransactionDetail";

type TransactionAmountProperties = {
	amount: number;
	convertedAmount?: number;
	returnedAmount?: number;
	currency: string;
	exchangeCurrency?: string;
	isMultiPayment?: boolean;
	isSent: boolean;
} & TransactionDetailProperties;

export const TransactionAmount: React.FC<TransactionAmountProperties> = ({
	amount,
	convertedAmount,
	returnedAmount,
	currency,
	exchangeCurrency,
	isMultiPayment,
	isSent,
	...properties
}: TransactionAmountProperties) => {
	const { t } = useTranslation();

	const renderModeIcon = () => {
		const modeIconName = isSent ? "Sent" : "Received";
		const tooltipContent = t(`TRANSACTION.${modeIconName.toUpperCase()}`);

		const modeCircleStyle = isSent
			? "border-theme-danger-100 text-theme-danger-500 dark:border-theme-danger-400 dark:text-theme-danger-400"
			: "border-theme-success-300 text-theme-success-600 dark:border-theme-success-600";

		return (
			<Tooltip content={tooltipContent}>
				<Circle className={modeCircleStyle} size="lg">
					<Icon name={modeIconName} size="lg" />
				</Circle>
			</Tooltip>
		);
	};

	const renderHintIcon = () => {
		if (!returnedAmount) {
			return;
		}

		const hintStyle = isSent
			? "bg-theme-danger-100 dark:bg-theme-danger-400"
			: "bg-theme-success-200 dark:bg-theme-success-600";

		return (
			<Tooltip content={t("TRANSACTION.HINT_AMOUNT", { amount: returnedAmount, currency })}>
				<div
					data-testid="TransactionAmount__Hint_Amount"
					className={cn("flex items-center -ml-1.5 px-2", hintStyle)}
				>
					<Icon name="HintSmall" size="sm" className="dark:text-white" />
				</div>
			</Tooltip>
		);
	};

	return (
		<TransactionDetail
			data-testid="TransactionAmount"
			label={isMultiPayment ? t("TRANSACTION.TOTAL_AMOUNT") : t("TRANSACTION.AMOUNT")}
			extra={renderModeIcon()}
			{...properties}
		>
			<Label color={isSent ? "danger" : "success"}>
				<div className="flex space-x-1">
					{renderHintIcon()}
					<AmountCrypto showSign ticker={currency} value={amount} isNegative={isSent} />
				</div>
			</Label>

			{!!exchangeCurrency && !!convertedAmount && (
				<Amount ticker={exchangeCurrency} value={convertedAmount} className="ml-2 text-theme-secondary-400" />
			)}
		</TransactionDetail>
	);
};
