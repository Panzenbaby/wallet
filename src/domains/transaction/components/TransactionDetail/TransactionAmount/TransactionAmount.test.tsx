import React from "react";
import { act, fireEvent, render, screen } from "utils/testing-library";

import { translations as transactionTranslations } from "../../../i18n";
import { TransactionAmount } from "./TransactionAmount";

describe("TransactionAmount", () => {
	it("should render", () => {
		const { container } = render(<TransactionAmount amount={1} currency="DARK" />);

		expect(container).toMatchSnapshot();
	});

	it("should render currency amount", () => {
		const { container } = render(<TransactionAmount amount={1} currency="DARK" />);

		expect(container).toHaveTextContent("1 DARK");
		expect(container).toMatchSnapshot();
	});

	it("should render converted currency amount", () => {
		const { container } = render(
			<TransactionAmount amount={1} convertedAmount={1} currency="DARK" exchangeCurrency="ARK" />,
		);

		expect(container).toHaveTextContent("1 DARK");
		expect(container).toHaveTextContent("1 ARK");
	});

	it.each([false, true])("should render label for multiple recipients", (isMultiPayment) => {
		const { container } = render(<TransactionAmount amount={1} currency="DARK" isMultiPayment={isMultiPayment} />);

		expect(container).toHaveTextContent(
			isMultiPayment ? transactionTranslations.TOTAL_AMOUNT : transactionTranslations.AMOUNT,
		);
		expect(container).toMatchSnapshot();
	});

	it.each(["Sent", "Received"])("should render '%s' icon", (type) => {
		const { container } = render(<TransactionAmount amount={1} currency="DARK" isSent={type === "Sent"} />);

		expect(container).toHaveTextContent(`${type.toLowerCase()}.svg`);
	});

	it.each(["Sent", "Received"])("should render info indicator for '%s'", (type) => {
		render(
			<TransactionAmount amount={2} returnedAmount={1} isMultiPayment currency="DARK" isSent={type === "Sent"} />,
		);

		expect(screen.getByTestId("TransactionAmount__Hint_Amount")).toBeInTheDocument();
		expect(screen.getByText("hint-small.svg")).toBeInTheDocument();

		act(() => {
			fireEvent.mouseEnter(screen.getByTestId("TransactionAmount__Hint_Amount"));
		});

		expect(screen.getByText("Excluding 1 DARK sent to itself")).toBeInTheDocument();
	});
});
