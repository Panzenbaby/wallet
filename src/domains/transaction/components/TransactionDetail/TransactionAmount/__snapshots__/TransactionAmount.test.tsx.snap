// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionAmount should render 1`] = `
<div>
  <div
    class="TransactionDetail__TransactionDetailStyled-r7sfme-0 hEJkEi no-ligatures"
    data-testid="TransactionAmount"
  >
    <div
      class="flex-1 space-y-2 w-40 whitespace-nowrap"
    >
      <div
        class="text-sm font-semibold text-theme-secondary-700 no-ligatures"
      >
        Amount
      </div>
      <div
        class="font-semibold"
      >
        <div
          class="Label-sc-4akkzw-0 lmoJAx"
          color="success"
        >
          <span
            data-testid="AmountCrypto"
          >
            + 1 DARK
          </span>
        </div>
      </div>
    </div>
    <div
      class="Circle__CircleWrapper-d5u41q-0 bWntUS border-theme-success-300 text-theme-success-600 dark:border-theme-success-600 ring-theme-background"
    >
      <div
        class="sc-bdfBwQ bcEVDZ"
        height="16"
        width="16"
      >
        <svg>
          received.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`TransactionAmount should render currency amount 1`] = `
<div>
  <div
    class="TransactionDetail__TransactionDetailStyled-r7sfme-0 hEJkEi no-ligatures"
    data-testid="TransactionAmount"
  >
    <div
      class="flex-1 space-y-2 w-40 whitespace-nowrap"
    >
      <div
        class="text-sm font-semibold text-theme-secondary-700 no-ligatures"
      >
        Amount
      </div>
      <div
        class="font-semibold"
      >
        <div
          class="Label-sc-4akkzw-0 lmoJAx"
          color="success"
        >
          <span
            data-testid="AmountCrypto"
          >
            + 1 DARK
          </span>
        </div>
      </div>
    </div>
    <div
      class="Circle__CircleWrapper-d5u41q-0 bWntUS border-theme-success-300 text-theme-success-600 dark:border-theme-success-600 ring-theme-background"
    >
      <div
        class="sc-bdfBwQ bcEVDZ"
        height="16"
        width="16"
      >
        <svg>
          received.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`TransactionAmount should render label for multiple recipients 1`] = `
<div>
  <div
    class="TransactionDetail__TransactionDetailStyled-r7sfme-0 hEJkEi no-ligatures"
    data-testid="TransactionAmount"
  >
    <div
      class="flex-1 space-y-2 w-40 whitespace-nowrap"
    >
      <div
        class="text-sm font-semibold text-theme-secondary-700 no-ligatures"
      >
        Amount
      </div>
      <div
        class="font-semibold"
      >
        <div
          class="Label-sc-4akkzw-0 lmoJAx"
          color="success"
        >
          <span
            data-testid="AmountCrypto"
          >
            + 1 DARK
          </span>
        </div>
      </div>
    </div>
    <div
      class="Circle__CircleWrapper-d5u41q-0 bWntUS border-theme-success-300 text-theme-success-600 dark:border-theme-success-600 ring-theme-background"
    >
      <div
        class="sc-bdfBwQ bcEVDZ"
        height="16"
        width="16"
      >
        <svg>
          received.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`TransactionAmount should render label for multiple recipients 2`] = `
<div>
  <div
    class="TransactionDetail__TransactionDetailStyled-r7sfme-0 hEJkEi no-ligatures"
    data-testid="TransactionAmount"
  >
    <div
      class="flex-1 space-y-2 w-40 whitespace-nowrap"
    >
      <div
        class="text-sm font-semibold text-theme-secondary-700 no-ligatures"
      >
        Total Amount
      </div>
      <div
        class="font-semibold"
      >
        <div
          class="Label-sc-4akkzw-0 lmoJAx"
          color="success"
        >
          <span
            data-testid="AmountCrypto"
          >
            + 1 DARK
          </span>
        </div>
      </div>
    </div>
    <div
      class="Circle__CircleWrapper-d5u41q-0 bWntUS border-theme-success-300 text-theme-success-600 dark:border-theme-success-600 ring-theme-background"
    >
      <div
        class="sc-bdfBwQ bcEVDZ"
        height="16"
        width="16"
      >
        <svg>
          received.svg
        </svg>
      </div>
    </div>
  </div>
</div>
`;
